### ISSUE TITLE
Random crash when idle

### ISSUE THREAD

**Issue by TheSunCat** [Nheko-Reborn/nheko#1602]
### Describe the bug

I left nheko running in the background and it ended up crashing.
I realize I have no debug symbols for nheko, so I'm not sure what can be done with this, but I am reporting the crash in case something is useful that I don't see.

### To Reproduce

I have no idea how to reproduce this, sorry! nheko was not focused and I had run it from a terminal

### What happened?

nheko suddenly closed

### Expected behavior

nheko should not crash

### Version

0.11.3-2b1844d4

### Operating system

Linux

### Installation method

Some repository (AUR, homebrew, distribution repository, PPA, etc)

### Qt version

6.6.0-2

### C++ compiler

gcc (GCC) 13.2.1 20230801

### Desktop Environment

Hyprland

### Did you use profiles?

- [ ] Profiles used?

### Relevant log output

```shell
[2023-10-30 18:18:06.692] [net] [error] Failed to retrieve event with id $vmWIkhQAgy3Tphl74C-xpvD1fLaRtaQ1jWjZzUTLJkg, which was requested to show the replyTo for event 
[2023-10-30 18:18:06.693] [mtx] [error] Invalid event type: [json.exception.out_of_range.403] key 'content' not found {
  "errcode": "M_NOT_FOUND",
  "error": "Event not found."
}
[2023-10-30 18:18:06.693] [net] [error] Failed to retrieve event with id $vmWIkhQAgy3Tphl74C-xpvD1fLaRtaQ1jWjZzUTLJkg, which was requested to show the replyTo for event 
[2023-10-30 18:18:06.693] [mtx] [error] Invalid event type: [json.exception.out_of_range.403] key 'content' not found {
  "errcode": "M_NOT_FOUND",
  "error": "Event not found."
}
[2023-10-30 18:18:06.693] [net] [error] Failed to retrieve event with id $vmWIkhQAgy3Tphl74C-xpvD1fLaRtaQ1jWjZzUTLJkg, which was requested to show the replyTo for event 
[2023-10-30 18:18:06.693] [mtx] [error] Invalid event type: [json.exception.out_of_range.403] key 'content' not found {
  "errcode": "M_NOT_FOUND",
  "error": "Event not found."
}
[2023-10-30 18:18:06.693] [net] [error] Failed to retrieve event with id $vmWIkhQAgy3Tphl74C-xpvD1fLaRtaQ1jWjZzUTLJkg, which was requested to show the replyTo for event 
[2023-10-30 18:19:53.837] [mtx] [warning] Error parsing events: Empty event type, {
  "content": {
    "hidden_event_types": [
      "m.reaction",
      "m.call.candidates",
      "m.call.negotiate",
      "",
      "m.call.select_answer",
      "m.room.power_levels",
      "m.room.member"
    ]
  },
  "type": "im.nheko.hidden_events"
}
Error: signal 11:
nheko(+0x62738d)[0x563fa40b338d]
/usr/lib/libc.so.6(+0x3e710)[0x7f40c416f710]
/usr/lib/libQt6Core.so.6(_ZNK7QObject6threadEv+0x4)[0x7f40c4ba51b4]
nheko(+0x2c411b)[0x563fa3d5011b]
nheko(+0x36c4e9)[0x563fa3df84e9]
nheko(+0x2f0532)[0x563fa3d7c532]
/usr/lib/libQt6Core.so.6(_ZN7QObject5eventEP6QEvent+0x247)[0x7f40c4ba5077]
/usr/lib/libQt6Widgets.so.6(_ZN19QApplicationPrivate13notify_helperEP7QObjectP6QEvent+0x4b)[0x7f40c5b7318b]
/usr/lib/libQt6Core.so.6(_ZN16QCoreApplication15notifyInternal2EP7QObjectP6QEvent+0x1d8)[0x7f40c4b626d8]
/usr/lib/libQt6Core.so.6(_ZN23QCoreApplicationPrivate16sendPostedEventsEP7QObjectiP11QThreadData+0x32b)[0x7f40c4b62a5b]
/usr/lib/libQt6Core.so.6(+0x399d24)[0x7f40c4d99d24]
/usr/lib/libglib-2.0.so.0(+0x59f69)[0x7f40c47b6f69]
/usr/lib/libglib-2.0.so.0(+0xb8327)[0x7f40c4815327]
/usr/lib/libglib-2.0.so.0(g_main_context_iteration+0x32)[0x7f40c47b5162]
/usr/lib/libQt6Core.so.6(_ZN20QEventDispatcherGlib13processEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE+0x84)[0x7f40c4d97934]
/usr/lib/libQt6Core.so.6(_ZN10QEventLoop4execE6QFlagsINS_17ProcessEventsFlagEE+0x19e)[0x7f40c4b6cc5e]
/usr/lib/libQt6Core.so.6(_ZN16QCoreApplication4execEv+0x88)[0x7f40c4b65178]
nheko(main+0x1898)[0x563fa3c53ec8]
/usr/lib/libc.so.6(+0x27cd0)[0x7f40c4158cd0]
/usr/lib/libc.so.6(__libc_start_main+0x8a)[0x7f40c4158d8a]
nheko(+0x1cae25)[0x563fa3c56e25]
Error: signal 6:
nheko(+0x62738d)[0x563fa40b338d]
/usr/lib/libc.so.6(+0x3e710)[0x7f40c416f710]
/usr/lib/libc.so.6(+0x8e83c)[0x7f40c41bf83c]
/usr/lib/libc.so.6(raise+0x18)[0x7f40c416f668]
nheko(+0x627438)[0x563fa40b3438]
/usr/lib/libc.so.6(+0x3e710)[0x7f40c416f710]
/usr/lib/libQt6Core.so.6(_ZNK7QObject6threadEv+0x4)[0x7f40c4ba51b4]
nheko(+0x2c411b)[0x563fa3d5011b]
nheko(+0x36c4e9)[0x563fa3df84e9]
nheko(+0x2f0532)[0x563fa3d7c532]
/usr/lib/libQt6Core.so.6(_ZN7QObject5eventEP6QEvent+0x247)[0x7f40c4ba5077]
/usr/lib/libQt6Widgets.so.6(_ZN19QApplicationPrivate13notify_helperEP7QObjectP6QEvent+0x4b)[0x7f40c5b7318b]
/usr/lib/libQt6Core.so.6(_ZN16QCoreApplication15notifyInternal2EP7QObjectP6QEvent+0x1d8)[0x7f40c4b626d8]
/usr/lib/libQt6Core.so.6(_ZN23QCoreApplicationPrivate16sendPostedEventsEP7QObjectiP11QThreadData+0x32b)[0x7f40c4b62a5b]
/usr/lib/libQt6Core.so.6(+0x399d24)[0x7f40c4d99d24]
/usr/lib/libglib-2.0.so.0(+0x59f69)[0x7f40c47b6f69]
/usr/lib/libglib-2.0.so.0(+0xb8327)[0x7f40c4815327]
/usr/lib/libglib-2.0.so.0(g_main_context_iteration+0x32)[0x7f40c47b5162]
/usr/lib/libQt6Core.so.6(_ZN20QEventDispatcherGlib13processEventsE6QFlagsIN10QEventLoop17ProcessEventsFlagEE+0x84)[0x7f40c4d97934]
/usr/lib/libQt6Core.so.6(_ZN10QEventLoop4execE6QFlagsINS_17ProcessEventsFlagEE+0x19e)[0x7f40c4b6cc5e]
/usr/lib/libQt6Core.so.6(_ZN16QCoreApplication4execEv+0x88)[0x7f40c4b65178]
nheko(main+0x1898)[0x563fa3c53ec8]
/usr/lib/libc.so.6(+0x27cd0)[0x7f40c4158cd0]
/usr/lib/libc.so.6(__libc_start_main+0x8a)[0x7f40c4158d8a]
nheko(+0x1cae25)[0x563fa3c56e25]
fish: Job 1, 'nheko' terminated by signal SIGABRT (Abort)
```

### Backtrace

```shell
Can't repro, so no backtrace for now, sorry!
```

**Comment by NepNep21** [comment]
This is likely the same issue I and someone in the matrix room also experienced, backtrace here:
```
Thread 1 (Thread 0x7ffff0b2f180 (LWP 35420) "nheko"):
#0  QObject::thread() const (this=0x30) at /usr/src/debug/qt6-base/qtbase-everywhere-src-6.6.0/src/corelib/kernel/qobject.cpp:1610
#1  0x0000555557f893d8 in EventStore::get(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::basic_string_view<char, std::char_traits<char> >, bool, bool) (this=0x30, id="$zViPHuxqcl9yVU__-TFmVvvlEnwhMd3se47N3bZXaZk", related_to="", decrypt=true, resolve_edits=true) at /tmp/nheko/src/timeline/EventStore.cpp:817
#2  0x0000555558084239 in TimelineModel::dataById(QString const&, int, QString const&) (this=0x0, id=..., role=0, relatedTo=...) at /tmp/nheko/src/timeline/TimelineModel.cpp:1044
#3  0x000055555800d63f in operator()(QList<int> const&) const (__closure=0x55555dd4df18, changedRoles=...) at /tmp/nheko/src/timeline/EventDelegateChooser.cpp:169
#4  0x000055555800dc19 in operator()(QModelIndex const&, QModelIndex const&, QList<int> const&) const (__closure=0x55555dd4df10, topLeft=..., bottomRight=..., changedRoles=...) at /tmp/nheko/src/timeline/EventDelegateChooser.cpp:220
#5  0x000055555800f8e2 in QtPrivate::FunctorCall<QtPrivate::IndexesList<0, 1, 2>, QtPrivate::List<const QModelIndex&, const QModelIndex&, const QList<int>&>, void, EventDelegateChooser::DelegateIncubator::setInitialState(QObject*)::<lambda(const QModelIndex&, const QModelIndex&, const QList<int>&)> >::call(struct {...} &, void **) (f=..., arg=0x55555ac5ddd0) at /usr/include/qt6/QtCore/qobjectdefs_impl.h:137
#6  0x000055555800f85b in QtPrivate::Functor<EventDelegateChooser::DelegateIncubator::setInitialState(QObject*)::<lambda(const QModelIndex&, const QModelIndex&, const QList<int>&)>, 3>::call<QtPrivate::List<QModelIndex const&, QModelIndex const&, QList<int> const&>, void>(struct {...} &, void *, void **) (f=..., arg=0x55555ac5ddd0) at /usr/include/qt6/QtCore/qobjectdefs_impl.h:339
#7  0x000055555800f756 in QtPrivate::QCallableObject<EventDelegateChooser::DelegateIncubator::setInitialState(QObject*)::<lambda(const QModelIndex&, const QModelIndex&, const QList<int>&)>, QtPrivate::List<const QModelIndex&, const QModelIndex&, const QList<int>&>, void>::impl(int, QtPrivate::QSlotObjectBase *, QObject *, void **, bool *) (which=1, this_=0x55555dd4df00, r=0x55555d06f6f0, a=0x55555ac5ddd0, ret=0x0) at /usr/include/qt6/QtCore/qobjectdefs_impl.h:522
#8  0x00007ffff4fa5077 in QObject::event(QEvent*) (this=0x55555d06f6f0, e=0x7ff768031fa0) at /usr/src/debug/qt6-base/qtbase-everywhere-src-6.6.0/src/corelib/kernel/qobject.cpp:1437
#9  0x00007ffff5f7318b in QApplicationPrivate::notify_helper(QObject*, QEvent*) (this=<optimized out>, receiver=0x55555d06f6f0, e=0x7ff768031fa0) at /usr/src/debug/qt6-base/qtbase-everywhere-src-6.6.0/src/widgets/kernel/qapplication.cpp:3290
#10 0x00007ffff4f626d8 in QCoreApplication::notifyInternal2(QObject*, QEvent*) (receiver=0x55555d06f6f0, event=event@entry=0x7ff768031fa0) at /usr/src/debug/qt6-base/qtbase-everywhere-src-6.6.0/src/corelib/kernel/qcoreapplication.cpp:1118
#11 0x00007ffff4f62a5b in QCoreApplication::sendEvent(QObject*, QEvent*) (event=0x7ff768031fa0, receiver=<optimized out>) at /usr/src/debug/qt6-base/qtbase-everywhere-src-6.6.0/src/corelib/kernel/qcoreapplication.cpp:1536
#12 QCoreApplicationPrivate::sendPostedEvents(QObject*, int, QThreadData*) (receiver=0x0, event_type=0, data=0x55555929a7f0) at /usr/src/debug/qt6-base/qtbase-everywhere-src-6.6.0/src/corelib/kernel/qcoreapplication.cpp:1898
#13 0x00007ffff5199d24 in QCoreApplication::sendPostedEvents(QObject*, int) (receiver=0x0, event_type=0) at /usr/src/debug/qt6-base/qtbase-everywhere-src-6.6.0/src/corelib/kernel/qcoreapplication.cpp:1757
#14 postEventSourceDispatch(GSource*, GSourceFunc, gpointer) (s=0x55555929c8e0) at /usr/src/debug/qt6-base/qtbase-everywhere-src-6.6.0/src/corelib/kernel/qeventdispatcher_glib.cpp:243
#15 0x00007ffff472bf19 in  () at /usr/lib/libglib-2.0.so.0
#16 0x00007ffff478a2b7 in  () at /usr/lib/libglib-2.0.so.0
#17 0x00007ffff472a112 in g_main_context_iteration () at /usr/lib/libglib-2.0.so.0
#18 0x00007ffff5197934 in QEventDispatcherGlib::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) (this=0x55555929ce10, flags=...) at /usr/src/debug/qt6-base/qtbase-everywhere-src-6.6.0/src/corelib/kernel/qeventdispatcher_glib.cpp:393
#19 0x00007ffff4f6cc5e in QEventLoop::processEvents(QFlags<QEventLoop::ProcessEventsFlag>) (flags=..., this=0x7fffffffddf0) at /usr/src/debug/qt6-base/qtbase-everywhere-src-6.6.0/src/corelib/kernel/qeventloop.cpp:100
#20 QEventLoop::exec(QFlags<QEventLoop::ProcessEventsFlag>) (this=0x7fffffffddf0, flags=...) at /usr/src/debug/qt6-base/qtbase-everywhere-src-6.6.0/src/corelib/kernel/qeventloop.cpp:182
#21 0x00007ffff4f65178 in QCoreApplication::exec() () at /usr/src/debug/qt6-base/qtbase-everywhere-src-6.6.0/src/corelib/global/qflags.h:74
#22 0x00005555584aaa6d in main(int, char**) (argc=1, argv=0x7fffffffe338) at /tmp/nheko/src/main.cpp:431
```

**Comment by deepbluev7** [comment]
There is a small chance I fixed this on master. Probably hard to confirm, but if it seems to not happen anymore, can you give some feedback? :)

**Comment by NepNep21** [comment]
It does actually seem to be fixed on 81cf29f92427c0c564861aa34bba632f98fad42e, haven't had any crashes since yesterday (they used to happen quite a few times per day).

**Comment by deepbluev7** [comment]
Neat, thank you!
